### PR TITLE
[Merged by Bors] - Update db schema to store "from" from the vestingSchedule added event

### PR DESCRIPTION
**Pull request by saadjhk**
## Issue
Need information of `from` on the frontend to determine whether a vesting schedule was initiated by bondedFinance pallet. 

## Description
I have updated vesting schedule entity and vesting schedule processor to support storing `from` key from the event. There are few typescript updates and fixes as well

### CODE REVIEW

**Review by vimukthi-git — APPROVED** [review]
LGTM

File: subsquid/schema.graphql [hunk 1]
```diff
@@ -92,6 +92,9 @@ type Schedule {
 
 type VestingSchedule @entity {
   id: ID!
+  "account that initiates the schedule"
+  from: String!
+  "random UUID"
```
Comment (andresmechali) [inline subsquid/schema.graphql:97]:
```suggestion
  "Chain event ID"
```

File: subsquid/schema.graphql [hunk 2]
```diff
@@ -92,6 +92,9 @@ type Schedule {
 
 type VestingSchedule @entity {
   id: ID!
+  "account that initiates the schedule"
+  from: String!
```
Comment (andresmechali) [inline subsquid/schema.graphql:96]: Given that we are adding `from` we can rename `beneficiary` to `to` to keep it consistent

**Review by andresmechali — CHANGES_REQUESTED** [review]
The changes look good. In `vestingTest.ts` you should also add `from` to the `assertVestingSchedule` and `assertVestingScheduleAddedEvent` functions

File: subsquid/db/migrations/1658391982828-Init.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ module.exports = class Init1657889647593 {
...
     await db.query(`CREATE INDEX "IDX_969a927080f5b6c81b79b40cd8" ON "pablo_transaction" ("pool_id") `)
     await db.query(`CREATE TABLE "bonded_finance_bond_offer" ("id" character varying NOT NULL, "event_id" text NOT NULL, "offer_id" text NOT NULL, "total_purchased" numeric NOT NULL, "beneficiary" text NOT NULL, CONSTRAINT "PK_1a7a97e3d57a4ac842dc2ef48ba" PRIMARY KEY ("id"))`)
-    await db.query(`CREATE TABLE "vesting_schedule" ("id" character varying NOT NULL, "event_id" text NOT NULL, "schedule_id" text NOT NULL, "beneficiary" text NOT NULL, "schedule" jsonb NOT NULL, CONSTRAINT "PK_4818b05532ed9058110ed5b5b13" PRIMARY KEY ("id"))`)
+    await db.query(`CREATE TABLE "vesting_schedule" ("id" character varying NOT NULL, "event_id" text NOT NULL, "from" text NOT NULL, "schedule_id" text NOT NULL, "beneficiary" text NOT NULL, "schedule" jsonb NOT NULL, CONSTRAINT "PK_4818b05532ed9058110ed5b5b13" PRIMARY KEY ("id"))`)
```
Comment (andresmechali) [inline subsquid/db/migrations/1658391982828-Init.js:14]: This needs to be updated for the last schema

**Comment by dzmitry-lahoda** [comment]
image

**Comment by saadjhk** [comment]
> image

@dzmitry-lahoda subsquid tests are fixed now.

**Review by andresmechali — APPROVED** [review]
LGTM 🚀

**Review by andor0 — APPROVED** [review]

File: subsquid/src/pabloProcessor.ts [hunk 4]
```diff
@@ -163,13 +162,12 @@ interface LiquidityAddedEvent {
...
-    const { who, poolId, baseAmount, quoteAmount, mintedLp } = event.asV2300;
-    return { who, poolId, baseAmount, quoteAmount, mintedLp };
-  } else {
-    const { who, poolId, baseAmount, quoteAmount, mintedLp } = event.asLatest;
+  if (event.isV2400) {
+    const { who, poolId, baseAmount, quoteAmount, mintedLp } = event.asV2400;
     return { who, poolId, baseAmount, quoteAmount, mintedLp };
   }
+  const { who, poolId, baseAmount, quoteAmount, mintedLp } = event.asLatest;
+  return { who, poolId, baseAmount, quoteAmount, mintedLp };
```
Comment (easteregg) [inline subsquid/src/pabloProcessor.ts:170]: I think in the end we can ignore the restructuring as it doesn't add much value here, so: 
 ```suggestion
 return event.isV2400 ?? event.asLatest;
```

Comment (easteregg) [reply subsquid/src/pabloProcessor.ts:170]: This can be used throughout all event getters.

File: subsquid/src/vestingProcessor.ts [hunk 5]
```diff
@@ -59,15 +59,17 @@ export async function processVestingScheduleAddedEvent(
...
+  const fromAccount = encodeAccount(from);
 
   const vestingSchedule = new VestingSchedule({
     id: randomUUID(),
+    from: fromAccount,
     eventId: ctx.event.id,
-    scheduleId: `${beneficiary}-${asset.toString()}`,
-    beneficiary,
+    scheduleId: `${toAccount}-${asset.toString()}`,
+    to: toAccount,
```
Comment (easteregg) [inline subsquid/src/vestingProcessor.ts:72]: is this the same as beneficiary? `to` is a bit ambiguous, unless we add a comment line explaining.

Comment (saadjhk) [reply subsquid/src/vestingProcessor.ts:72]: `scheduleId` is not the same as beneficiary, it is a value generated here to create unique ids for vesting schedules as they did not have them previously on the pallet

**Review by easteregg — APPROVED** [review]

**Review by andresmechali — APPROVED** [review]
🚀

**Review by vimukthi-git — APPROVED** [review]

**Pull request closed without merging**